crates/static-analysis-kernel/src/analysis/ddsa_lib/common.rs

```diff
@@ -232,17 +239,52 @@ pub fn set_key_value<G>(
     v8_object.set(scope, v8_key.into(), v8_value);
 }
 
-/// Attaches the provided `v8_item` to the [`v8::Context::global`] with identifier `name`, overwriting
-/// any previous value.
-pub fn attach_as_global<'s, T>(
-    scope: &mut HandleScope<'s>,
-    v8_item: impl v8::Handle<Data = T>,
-    name: &str,
-) where
-    v8::Local<'s, v8::Value>: From<v8::Local<'s, T>>,
-{
-    let v8_local = v8::Local::new(scope, v8_item);
-    let global = scope.get_current_context().global(scope);
-    let v8_key = v8_string(scope, name);
-    global.set(scope, v8_key.into(), v8_local.into());
+/// Creates a `v8::Global` [`UnboundScript`](v8::UnboundScript) from the given code.
+pub fn compile_script(
+    scope: &mut HandleScope,
+    code: &str,
+) -> Result<v8::Global<v8::UnboundScript>, DDSAJsRuntimeError> {
+    let code_str = v8_string(scope, code);
+    let tc_scope = &mut v8::TryCatch::new(scope);
+    let script_result = v8::Script::compile(tc_scope, code_str, None);
+
+    let script = script_result.ok_or_else(|| {
+        let exception = tc_scope
+            .exception()
+            .expect("return value should only be `None` if an error was caught");
+        let reason = exception.to_rust_string_lossy(tc_scope);
+        tc_scope.reset();
+        DDSAJsRuntimeError::Interpreter { reason }
+    })?;
+
+    let unbound_script = script.get_unbound_script(tc_scope);
+    Ok(v8::Global::new(tc_scope, unbound_script))
+}
```

Comment:
It seems this function is only used for tests purposes. Should we either?

1. Only enable it for tests?
2. Put a comment to explain this?